tests/Microsoft.Identity.Web.Test/WebAppExtensionsTests.cs

```diff
@@ -1,32 +1,493 @@
 ﻿// Copyright (c) Microsoft Corporation. All rights reserved.
 // Licensed under the MIT License.
 
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Security.Claims;
+using System.Threading.Tasks;
+using Microsoft.AspNetCore.Authentication;
+using Microsoft.AspNetCore.Authentication.Cookies;
 using Microsoft.AspNetCore.Authentication.OpenIdConnect;
+using Microsoft.AspNetCore.Http;
+using Microsoft.Extensions.Configuration;
+using Microsoft.Extensions.Configuration.Memory;
+using Microsoft.Extensions.DependencyInjection;
+using Microsoft.Extensions.DependencyInjection.Extensions;
+using Microsoft.Extensions.Options;
+using Microsoft.Extensions.Primitives;
+using Microsoft.Identity.Client;
+using Microsoft.Identity.Web.Resource;
 using Microsoft.Identity.Web.Test.Common;
+using Microsoft.Identity.Web.Test.Common.TestHelpers;
+using Microsoft.IdentityModel.Protocols.OpenIdConnect;
+using NSubstitute;
+using NSubstitute.Extensions;
 using Xunit;
 
 namespace Microsoft.Identity.Web.Test
 {
     public class WebAppExtensionsTests
     {
+        private const string _configSectionName = "AzureAd-Custom";
+        private const string _oidcScheme = "OpenIdConnect-Custom";
+        private const string _cookieScheme = "Cookies-Custom";
+        private IConfigurationSection _configSection;
+        private readonly Action<ConfidentialClientApplicationOptions> _configureAppOptions = (options) => { };
+        private readonly Action<OpenIdConnectOptions> _configureOidcOptions = (options) => {
+            options.ClientId = TestConstants.ClientId;
+        };
+        private Action<MicrosoftIdentityOptions> _configureMsOptions = (options) => {
+            options.Instance = TestConstants.AadInstance;
+            options.TenantId = TestConstants.TenantIdAsGuid;
+            options.ClientId = TestConstants.ClientId;
+        };
+
+        public WebAppExtensionsTests()
+        {
+            _configSection = GetConfigSection(_configSectionName);
+        }
+
+        [Theory]
+        [InlineData(true)]
+        [InlineData(false)]
+        public void AddSignIn_WithConfigName(bool useServiceCollectionExtension)
+        {
+            var configMock = Substitute.For<IConfiguration>();
+            configMock.Configure().GetSection(_configSectionName).Returns(_configSection);
+              
+            var services = new ServiceCollection();
+            services.AddDataProtection();
+
+            if (useServiceCollectionExtension)
+                services.AddSignIn(configMock, _configSectionName, _oidcScheme, _cookieScheme, false);
```

Comment:
can you add the { }...with the analyzers you added, this becomes a warning, which I've fixed in the branch i pushed to your other branch. thx.

Reply:
why not merge the other branch?

Reply:
I don't know. It's Peter's branch. I need you to look at the doc warnings though. I'm not sure what we want to do w/some of the xml stuff. there are 400+ warnings still in that branch. I think majority are docs now...we can discuss.

Reply:
Will do.